pkg/eventcounter/api/api.go

```diff
@@ -535,6 +536,10 @@ func (s *eventCounterService) countUniqueUser(
 		err = append(err, e)
 		return
 	}
+	// Set expiration time for pfmerge key in case delete fails
+	if _, e := s.evaluationCountCacher.ExpireKey(key, 10*time.Minute); e != nil {
+		s.logger.Error("Failed to set expiration for pfmerge key", zap.Error(e))
+	}
```

Comment:
Sorry for my bad explanation. I thought the pfmerge command would also accept a parameter to set the expiration, so it could be done using only one request so that we won't need the deleting implementation.
This will introduce latency, and if the set expiration fails, the delete will probably fail, too (If Redis is down or unstable for a moment) because both requests are made right after the other.

While creating a new kind for pfmerge keys will fix the error in the redis counter deleted, it still could leave the undeleted key in the Redis forever in case the deleting request fails. We need a better way to avoid this.

Reply:
It makes sense for me!
Let me implement expiration params in pfmerge function directly.